fusion_toolbox/shot.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+class Shot:
+    """Shot class to store data from a shot."""
+    def __init__(self, path: str, signals:list = []):
+        self.path = path
+        self.signals = signals
+        self.data = {} # Dictionary of signals. Each signal is a dictionary with keys "time" and "value"
+        self.tokamak = None
+        self.shot_id = None
+        self.events = {}
+        
+        #For now, default to loading from csv file
+        self.load_csv()
+        
+
+    def load_csv(self):
```

Comment:
```suggestion
    @classmethod
    def from_csv(self, path=None):
```
This function feels a bit awkward, because the user has no control over the `load_csv` method (i.e. no arguments to pass). If this is only used in the constructor then it's probably not worth having as a method on it's own. The way `from_file` methods typically work is as class methods to give the user the option of specifying a path.